binder/src/main/java/io/grpc/binder/internal/BinderTransportSecurity.java

```diff
@@ -103,7 +106,7 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
       // Most SecurityPolicy will have synchronous implementations that provide an
       // immediately-resolved Future. In that case, short-circuit to avoid unnecessary allocations
       // and asynchronous code if the authorization result is already present.
-      if (!authStatusFuture.isDone()) {
+      if (!authStatusFuture.isDone() || authStatusFuture.isCancelled()) {
```

Comment:
Couldn't you just catch ExecutionException | CancellationException on line 116 below instead? These futures should never be cancelled, so that would be an INTERNAL just like any other type of failure.

Reply:
Good call. Moved to your suggestion.